From ea8601576b3692a5b1ab9a7208a4a708f24a7001 Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Wed, 8 Jul 2020 09:04:57 +0200
Subject: [PATCH 2/2] Bind to OpenJDK 21 for runtime

Forwarded: not-needed
---
 apache-maven/src/bin/mvn | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/apache-maven/src/bin/mvn b/apache-maven/src/bin/mvn
index 38b036af1d..d773ab6c2c 100755
--- a/apache-maven/src/bin/mvn
+++ b/apache-maven/src/bin/mvn
@@ -28,21 +28,7 @@
 #   MAVEN_SKIP_RC   (Optional) Flag to disable loading of mavenrc files.
 # -----------------------------------------------------------------------------
 
-if [ -z "$MAVEN_SKIP_RC" ] ; then
-
-  if [ -f /usr/local/etc/mavenrc ] ; then
-    . /usr/local/etc/mavenrc
-  fi
-
-  if [ -f /etc/mavenrc ] ; then
-    . /etc/mavenrc
-  fi
-
-  if [ -f "$HOME/.mavenrc" ] ; then
-    . "$HOME/.mavenrc"
-  fi
-
-fi
+export JAVA_HOME=/usr/lib/jvm/java-21
 
 # OS specific support. $var _must_ be set to either true or false.
 cygwin=false;
-- 
2.46.0

