From ebb8f31b84299d5564e26ecb22845133819c9aae Mon Sep 17 00:00:00 2001
From: Marian Koncek <mkoncek@redhat.com>
Date: Mon, 17 Aug 2020 13:39:11 +0200
Subject: [PATCH] Remove usage of ArchiveStreamFactory

Forwarded: not-needed
---
 .../archivers/zip/ZipSplitReadOnlySeekableByteChannel.java     | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/main/java/org/apache/commons/compress/archivers/zip/ZipSplitReadOnlySeekableByteChannel.java b/src/main/java/org/apache/commons/compress/archivers/zip/ZipSplitReadOnlySeekableByteChannel.java
index 956887d6..76462ab7 100644
--- a/src/main/java/org/apache/commons/compress/archivers/zip/ZipSplitReadOnlySeekableByteChannel.java
+++ b/src/main/java/org/apache/commons/compress/archivers/zip/ZipSplitReadOnlySeekableByteChannel.java
@@ -35,7 +35,6 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import org.apache.commons.compress.archivers.ArchiveStreamFactory;
 import org.apache.commons.compress.utils.FileNameUtils;
 import org.apache.commons.compress.utils.MultiReadOnlySeekableByteChannel;
 
@@ -98,7 +97,7 @@ public static SeekableByteChannel buildFromLastSplitSegment(final File lastSegme
      */
     public static SeekableByteChannel buildFromLastSplitSegment(final Path lastSegmentPath) throws IOException {
         final String extension = FileNameUtils.getExtension(lastSegmentPath);
-        if (!extension.equalsIgnoreCase(ArchiveStreamFactory.ZIP)) {
+        if (!extension.equalsIgnoreCase("zip")) {
             throw new IllegalArgumentException("The extension of last ZIP split segment should be .zip");
         }
 
-- 
2.40.1

